### DISCUSSION TITLE
Set global authorization header to $fetch that is available on both client and server?

### DISCUSSION THREAD

**Discussion by danielrjames** [nuxt/nuxt#15961] [Questions] [answered]
Hi, I was wondering if $fetch allows a global authorization header to be set that is available on both client and server (such as saveToken in the nuxt/axios module for nuxt 2)?

I've seen examples of the header being set on an individual request, but nothing globally, such as in a store after a user login. If this doesn't already exist, I think this would be a critical feature needed for widescale adoption.

Thank you again for all the hard work. Excited for the official release!

**Comment by Diizzayy** [comment, marked as answer]
@danielrjames This feature isn't currently supported out-the-box but it is very easily achieved.

`./nuxt-app/composables/useProfile.ts`
```ts
export const useProfile = () => {
  const token = useState<string>("token");

  const setToken = (value: string) => {
    token.value = value;
  };

  return { id: null, name: null, token, setToken };
};
```

`./nuxt-app/composables/useFetchAuth.ts`
```ts
import { FetchOptions } from "ohmyfetch";

export const useFetchAuth = (url: string, opts?: FetchOptions) => {
  const { token } = useProfile();

  const headers: HeadersInit = {
    ...(opts?.headers || {}),
    ...(token && { Authorization: `Bearer ${token.value}` }),
  };

  return $fetch(url, { ...opts, headers });
};
```

Given the code above:
- You can set an auth token by `useProfile().setToken('secret_token')`
- Any calls to `useFetchAuth()` in your app (pages, components, composables, etc...) will be sent with the proper Authorization header once the token has been set.

**Reply by nowshad-hossain-rahat** [reply]
worked perfectly

**Comment by ThinkinDragon** [comment]
is there any way to use it in plugin ?

**Comment by kaboume** [comment]
Thank you for the answer.
It's still annoying to replace in all the code useFetch by useFetchAuth.

Is a setting somewhere or a plugin planned to integrate this?
